include/seastar/core/internal/pollable_fd.hh

```diff
@@ -68,13 +68,13 @@ public:
     /// blocked and it didn't return the short read/write. We anticipate that
     /// the next I/O will also be non-blocking and will not return EAGAIN.
     /// But the speculation is invalidated once it is "used" by
-    /// \c take_speculation()
-    void speculate_epoll(int events) { events_known |= events; }
+    /// \c revoke_speculation()
+    void take_speculation(int events) { events_known |= events; }
     /// Check whether we speculate specified I/O is possible on the fd,
     /// invalidate the speculation if it matches with all specified \c events.
     ///
     /// \return true if the current speculation includes all specified events
-    bool take_speculation(int events) {
+    bool revoke_speculation(int events) {
```

Comment:
It's not just revoking, it's also returning the previous value.

Let's rename it to check_and_revoke_speculation() so an `if` statement makes sense. Something like `if (revoke_speculation)` doesn't.

Reply:
It not returns the previous value, but returns whether `events` exists, or whether the revocation is actually performed, just like what the STL `std::unordered_map::emplace` operation does, so I thought It would be okay to omit `check` in the name.

But I agree with what you said, the name does not reflect the `check` actually, making it weird when used in `if`. Readability is important and I'll change it.